### PR TITLE
Address PR comments in #74

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Other

**Required reviews:** 
-  1

**What this does:**
- Avoids `unwrap()`, and handles errors gracefully instead
- Replaces `PartialEq` impl. with a `derive` macro
- Also throwing in a commit to address changes in the new Rust `stable` (1.7.2)

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
- Based on #74 - Please review & merge it first
- The CI failure is caused by MSRV incompatibility. To be fixed.

### CODE REVIEW

**Comment by KPrasch** [comment]
rebased @ 43f943e80b844768d27e6643976133d9e9e7293d

File: nucypher-core/src/access_control.rs [hunk 1]
```diff
@@ -22,6 +21,8 @@ pub struct AuthenticatedData {
     pub conditions: Option<Conditions>,
 }
 
+impl Eq for AuthenticatedData {}
```
Comment (fjarri) [inline nucypher-core/src/access_control.rs:24]: Does it work if you just add it to the list in `#[derive]`?

Comment (piotr-roslaniec) [reply nucypher-core/src/access_control.rs:24]: Not quite. We'd also need to `#[derive(Eq)]` for `DkgPublicKey` in `ferveo`

**Review by manumonti — APPROVED** [review]
Not a rust developer, but looks good. ✌️

File: nucypher-core-wasm/src/lib.rs [hunk 2]
```diff
@@ -1101,7 +1101,7 @@ impl ReencryptionResponse {
 
         Ok(Self(nucypher_core::ReencryptionResponse::new(
             signer.as_ref(),
-            backend_capsules.iter().zip(backend_vcfrags.into_iter()),
+            backend_capsules.iter().zip(backend_vcfrags),
```
Comment (derekpierre) [inline nucypher-core-wasm/src/lib.rs:1104]: @piotr-roslaniec Let's be careful changing old PRE code unless necessary.

Is it that `into_iter()` no longer works, or is `zip(...)` just best practice?

**Pull request merged**